Sources/SourceKit/sourcekitd/SwiftLanguageServer.swift

```diff
@@ -42,40 +47,16 @@ public final class SwiftLanguageServer: LanguageServer {
 
   /// Creates a language server for the given client using the sourcekitd dylib at the specified path.
   public init(client: Connection, sourcekitd: AbsolutePath, buildSystem: BuildSystem, clientCapabilities: ClientCapabilities, onExit: @escaping () -> Void = {}) throws {
-
+    self.client = client
     self.sourcekitd = try SwiftSourceKitFramework(dylib: sourcekitd)
     self.buildSystem = buildSystem
     self.clientCapabilities = clientCapabilities
     self.documentManager = DocumentManager()
     self.onExit = onExit
-    super.init(client: client)
-  }
-
-  public override func _registerBuiltinHandlers() {
-
-    _register(SwiftLanguageServer.initialize)
-    _register(SwiftLanguageServer.clientInitialized)
-    _register(SwiftLanguageServer.cancelRequest)
-    _register(SwiftLanguageServer.shutdown)
-    _register(SwiftLanguageServer.exit)
-    _register(SwiftLanguageServer.didChangeConfiguration)
-    _register(SwiftLanguageServer.openDocument)
-    _register(SwiftLanguageServer.closeDocument)
-    _register(SwiftLanguageServer.changeDocument)
-    _register(SwiftLanguageServer.willSaveDocument)
-    _register(SwiftLanguageServer.didSaveDocument)
-    _register(SwiftLanguageServer.completion)
-    _register(SwiftLanguageServer.hover)
-    _register(SwiftLanguageServer.documentSymbolHighlight)
-    _register(SwiftLanguageServer.foldingRange)
-    _register(SwiftLanguageServer.symbolInfo)
-    _register(SwiftLanguageServer.documentSymbol)
-    _register(SwiftLanguageServer.documentColor)
-    _register(SwiftLanguageServer.colorPresentation)
-    _register(SwiftLanguageServer.codeAction)
   }
 
-  func publishDiagnostics(
+  /// Should be called on self.queue.
+  func queue_publishDiagnostics(
```

Comment:
I'm not a big fan of this naming convention, because it is likely to get lost if the particular queue changes or gets renamed. I'd prefer to just use a leading underscore and leave the queue-bit for the documentation.

Reply:
Done